api/__tests__/routes/region.test.ts

```diff
@@ -0,0 +1,357 @@
+                        data: newRegion,
+                        expectedResponse: forbiddenResponse,
+                        statusCode: 403,
+                    });
+                });
+
+                test("Cannot reach DELETE /region/:id", async () => {
+                    await runner.delete({
+                        url: "/region/1",
+                        statusCode: 403,
+                    });
+                });
+            });
+
+            describe("Cannot reach any path as a student", () => {
+                beforeEach(() => {
+                    runner.authLevel(AuthenticationLevel.STUDENT);
+                });
+
+                test("Cannot reach GET /region", async () => {
+                    await runner.get({
+                        url: "/region",
+                        expectedData: [forbiddenResponse],
+                        statusCode: 403,
+                    });
+                });
+
+                test("Cannot reach POST /region", async () => {
+                    await runner.post({
+                        url: "/region",
+                        data: newRegion,
+                        expectedResponse: forbiddenResponse,
+                        statusCode: 403,
+                    });
+                });
+
+                test("Cannot reach PATCH /region/:id", async () => {
+                    await runner.patch({
+                        url: "/region/1",
+                        data: newRegion,
+                        expectedResponse: forbiddenResponse,
+                        statusCode: 403,
+                    });
+                });
+
+                test("Cannot reach DELETE /region/:id", async () => {
+                    await runner.delete({
+                        url: "/region/1",
+                        statusCode: 403,
+                    });
+                });
+            });
+
+            describe("The requested path must exist", () => {
+                beforeEach(() => {
+                    runner.authLevel(AuthenticationLevel.ADMINISTRATOR);
+                });
+
+                test("Find a nonexistent region", async () => {
+                    await runner.get({
+                        url: "/region/0",
+                        expectedData: [notFoundResponse],
+                        statusCode: 404,
+                    });
+                });
+
+                test("Find a nonexistent region", async () => {
+                    await runner.get({
+                        url: "/region/-1",
+                        expectedData: [notFoundResponse],
+                        statusCode: 404,
+                    });
+                });
+
+                test("Update a nonexistent region", async () => {
+                    await runner.get({
+                        url: "/region/0",
+                        expectedData: [notFoundResponse],
+                        statusCode: 404,
+                    });
+                });
```

Comment:
Hetzelfde comment hier als in mijn review van Round tests. Ofwel neem deze testen samen, ofwel noem ze anders.